ouroboros-consensus/docs/diffusion-pipelining-design/block-diffusion-pipelining.tex

```diff
@@ -0,0 +1,554 @@
+  \draw (0,0) rectangle (2,1);
+  \draw[fill=lightgray] (2,0) rectangle (3,1);
+  \draw (3,0) rectangle (7,1);
+  \draw[fill=lightgray] (7,0) rectangle (13,1);
+  \draw (13,0) rectangle (14,1);
+  \draw (4, -0.5) node[anchor=north east] {Intermediate relay nodes};
+\end{scope}
+
+\begin{scope}[xshift=27cm, yshift=-6cm]
+  \draw (0,0) rectangle (2,1);
+  \draw[fill=lightgray] (2,0) rectangle (3,1);
+  \draw (3,0) rectangle (7,1);
+  \draw[fill=lightgray] (7,0) rectangle (13,1);
+  \draw (13,0) rectangle (14,1);
+\end{scope}
+
+\begin{scope}[xshift=34cm, yshift=-8cm]
+  \draw (0,0) rectangle (2,1);
+  \draw[fill=lightgray] (2,0) rectangle (3,1);
+  \draw (3,0) rectangle (7,1);
+  \draw[fill=lightgray] (7,0) rectangle (13,1);
+  \draw (0, 0.5) node[anchor=east] {Last block producer in path};
+\end{scope}
+
+\draw[->] (7, 2) -- (50, 2);
+\draw     (23.5, 2) node[anchor=south] {Time};
+
+\end{scope}
+\end{tikzpicture}
+\end{center}
+
+\pagebreak
+
+\section{Risk analysis}
+Before we analyse the risks it is important to be precise about which validations
+will still be executed as normal, and which would be deferred. The following are
+validations which should still be done:
+
+\begin{enumerate} 
+  \item The hash validation to ensure that the block body content matches the header;
+  \item Nodes relaying blocks would only relay a single block in advance of validating the block body; and
+  \item Block producers would still validate the block body before creating a new block.
+\end{enumerate}
+
+\emph{Thus, the only validation which is deferred is that on the block body content.}
+
+\section{Analysis of DoS risks}
+The immediate concern is whether executing less prompt validation would
+create any new opportunities for denial-of-service attacks. In general the reason
+we take a very conservative approach to chain diffusion -- in which we validate
+fully and promptly -- is to avoid denial-of-service attacks. The general
+principle has been that whilst we have found ways to bound the amount of work
+honest nodes need to do for valid block data, it is substantially harder to
+bound work if we propagate invalid data.
+
+That said, the goal is not to prevent invalid data from being relayed over the
+network graph. The goal is to prevent honest nodes from being required to do an
+unbounded amount of work, and indeed more stringently, to prevent honest nodes
+from having to do more work than the dishonest nodes engaged in an attack. Thus,
+if we can relax or defer validation without increasing the work compared with the
+normal valid case, then the risk may be acceptable. In particular, we need to avoid any
+situations whereby previously we executed a single expensive step, and now we would do many.
+
+The combination of checks 1.~and 2.~above should mean that the possibly-invalid
+block is one which was signed by the block producer (KES validation), who (in-turn) was
+authorised to create a block in this slot (VRF verification). This should mean that
```

Comment:
> This should mean that there cannot be a greater number of invalid blocks, in total, than in the case of valid blocks.

I'm not sure I follow this sentence. I'm guessing it's the "number of" and/or "in total" part I'm not grokking.

The proposal doesn't create new opportunities for minting invalid blocks. But it does -- for the first time ever -- allow honest nodes to transmit invalid blocks.

So, what is being counted in the words "greater number"?